src/main/java/bio/terra/service/snapshot/flight/create/AddDataAccessControlsStep.java

```diff
@@ -0,0 +1,37 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.service.job.DefaultUndoStep;
+import bio.terra.service.snapshot.SnapshotService;
+import bio.terra.service.snapshot.flight.SnapshotWorkingMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+public class AddDataAccessControlsStep extends DefaultUndoStep {
+  private final SnapshotService snapshotService;
+  private final AuthenticatedUserRequest userRequest;
+  private final List<String> dataAccessControlGroups;
+
+  public AddDataAccessControlsStep(
+      SnapshotService snapshotService,
+      AuthenticatedUserRequest userRequest,
+      List<String> dataAccessControlGroups) {
+    this.snapshotService = snapshotService;
+    this.userRequest = userRequest;
+    this.dataAccessControlGroups = dataAccessControlGroups;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    snapshotService.addSnapshotDataAccessControls(
```

Comment:
This service call ends up starting another flight. Would it work to instead insert the DAC steps directly into this flight? That could be better, as the snapshot is still being created while this flight is executing, so the lock step in `SnapshotAddDataAccessControlsFlight` could fail. If it's called directly, then locking wouldn't be necessary.

It might also be good to tie the success of the DAC setting with the snapshot creation process itself, to avoid creating a snapshot that doesn't have its DAC set properly.

I think that this can be done by adding the steps `CreateSnapshotGroupConstraintPolicyStep`, `AddSnapshotAuthDomainStep` and `JournalRecordUpdateEntryStep` directly to the `SnapshotCreateFlight`.